src/components/data-pane/field-list.tsx

```diff
@@ -1,49 +1,71 @@
 import * as React from 'react';
 import * as CSSModules from 'react-css-modules';
 import {connect} from 'react-redux';
+import * as TetherComponent from 'react-tether';
 import * as styles from './field-list.scss';
 
+import {ExpandedType} from 'compassql/build/src/query/expandedtype';
+import {PrimitiveType, Schema} from 'compassql/build/src/schema';
+import {DatasetSchemaChangeFieldType} from '../../actions/dataset';
 import {ActionHandler, createDispatchHandler} from '../../actions/redux-action';
 import {SHELF_FIELD_AUTO_ADD, ShelfFieldAutoAdd} from '../../actions/shelf';
 import {FieldParentType} from '../../constants';
 import {State} from '../../models/index';
 import {ShelfFieldDef} from '../../models/shelf/encoding';
 import {getPresetWildcardFields, getSchemaFieldDefs} from '../../selectors';
 import {Field} from '../field';
+import {TypeChanger} from './type-changer';
 
 
-export interface FieldListProps extends ActionHandler<ShelfFieldAutoAdd> {
+export interface FieldListProps extends ActionHandler<ShelfFieldAutoAdd | DatasetSchemaChangeFieldType> {
   fieldDefs: ShelfFieldDef[];
+  schema: Schema;
 }
 
-class FieldListBase extends React.PureComponent<FieldListProps, {}> {
+export interface FieldListState {
+  selectedField: string;
+}
+
+class FieldListBase extends React.PureComponent<FieldListProps, FieldListState> {
 
   constructor(props: FieldListProps) {
     super(props);
 
     // Bind - https://facebook.github.io/react/docs/handling-events.html
     this.onAdd = this.onAdd.bind(this);
+    this.state = {
+      selectedField: null
+    };
   }
 
   public render() {
-    const {fieldDefs} = this.props;
-
+    const {fieldDefs, schema} = this.props;
     const fieldItems = fieldDefs.map(fieldDef => {
+      let primitiveType;
+      if (typeof fieldDef.field === 'string') {
+        primitiveType = schema.primitiveType(fieldDef.field);
+      }
+      const hideTypeChanger = (primitiveType !== PrimitiveType.NUMBER && primitiveType !== PrimitiveType.INTEGER)
```

Comment:
Instead of hard coding primitiveType check here, maybe call `getTypes` and check if there are more than one valid types (which mean we can change?).